package fetch.cfg.device;

import "scanner3d.proto";
import "stage.proto";
import "vibratome.proto";
import "workers.proto";
import "file.proto";
import "tasks.proto";

message FieldOfViewGeometry
{
  optional float x_size_um        = 1 [ default = 100.0];
  optional float y_size_um        = 2 [ default = 100.0];
  optional float z_size_um        = 3 [ default = 100.0];     // (a little) redundant with stack thickness - scanner3d.zpiezo.(um_max-um_min)
  optional float x_overlap_um     = 4 [ default = 10.0 ];     //            seperate here because we aren't always doing tiled acquisitions
  optional float y_overlap_um     = 5 [ default = 10.0 ];
  optional float z_overlap_um     = 6 [ default = 10.0 ];
  optional float rotation_radians = 7 [ default = 0.0  ];
}

message Microscope
{
  required Scanner3D                   scanner3d             = 1;
  required Stage                       stage                 = 2;
  optional Vibratome                   vibratome             = 3;
  optional FieldOfViewGeometry         fov                   = 4;
  required worker.ResonantWrap         resonant_wrap         = 5;
  optional worker.ResonantUnwarp       resonant_unwarp       = 13;
  required worker.HorizontalDownsample horizontal_downsample = 6;
  required worker.FrameAverage         frame_average         = 7;
  required FileSeries                  file_series           = 8;
  optional string                      file_prefix           = 9 [default="default"];
  optional string                      stack_extension       =10 [default=".tif"];
  optional string                      config_extension      =11 [default=".microscope"];
  optional string                      metadata_extension    =12 [default=".acquisition"];
  required tasks.AutoTile              autotile              =14;
}
